tests/test_routes.py

```diff
@@ -0,0 +1,247 @@
+
+    assert response.status_code == 200
+    assert len(response_body) == 2
+    assert response_body[0] == {
+        "id": 1,
+        "name": "Mercury",
+        "description": "The smallest planet; pitted and streaky, brownish-gray",
+        "distance_from_the_sun": 0.39
+    }
+    assert response_body[1] == {
+        "id": 2,
+        "name": "Venus",
+        "description": "A fireball with temperatures hot enough to melt lead, covered in thick clouds",
+        "distance_from_the_sun": 0.72
+    }
+
+
+def test_get_all_planets_with_name_query_matching_none(client, two_saved_planets):
+    data = {"name": "Eros"}
+    response = client.get("/planets", query_string=data)
+    response_body = response.get_json()
+
+    assert response.status_code == 200
+    assert response_body == []
+
+
+def test_get_all_planets_with_name_query_matching_one(client, two_saved_planets):
+    data = {"name": "Venus"}
+    response = client.get("/planets", query_string=data)
+    response_body = response.get_json()
+
+    assert response.status_code == 200
+    assert len(response_body) == 1
+    assert response_body[0] == {
+        "id": 2,
+        "name": "Venus",
+        "description": "A fireball with temperatures hot enough to melt lead, covered in thick clouds",
+        "distance_from_the_sun": 0.72
+    }
+
+
+def test_get_one_planet(client, two_saved_planets):
+    response = client.get("/planets/1")
+    response_body = response.get_json()
+
+    assert response.status_code == 200
+    assert response_body == {
+        "id": 1,
+        "name": "Mercury",
+        "description": "The smallest planet; pitted and streaky, brownish-gray",
+        "distance_from_the_sun": 0.39
+    }
+
+
+def test_get_one_planet_missing_record(client, two_saved_planets):
+    response = client.get("/planets/100")
+    response_body = response.get_json()
+
+    assert response.status_code == 404
+    assert response_body == {"message": "Planet 100 not found"}
+
+
+def test_get_one_planet_invalid_id(client, two_saved_planets):
+    response = client.get("planets/faraway")
+    response_body = response.get_json()
+
+    assert response.status_code == 400
+    assert response_body == {"message": "Planet faraway invalid"}
+
+
+def test_create_one_planet(client):
+    test_data = {
+        "name": "Earth",
+        "description": "A beautiful blue-green planet with life-supporting atmosphere",
+        "distance_from_the_sun": 1.00
+    }
+    response = client.post("/planets", json=test_data)
+    response_body = response.get_data(as_text=True)
+
+    assert response.status_code == 201
+    assert response_body == "Planet Earth successfully created"
+
+
+def test_create_one_planet_no_name(client):
+    test_data = {
+        "description": "A beautiful blue-green planet with life-supporting atmosphere",
+        "distance_from_the_sun": 1.00
+    }
+
+    with pytest.raises(KeyError, match="name"):
+        response = client.post("/planets", json=test_data)
```

Comment:
👀 The `client` test layer is lying a little bit here. If a `KeyError` is raised in our endpoint, it's actually going to result in a 500 Internal Server Error being sent back. We'd like to avoid this and let the caller know that it was theor fault the request couldn't be completed, which we would do by catching the `KeyError` in the route and sending back a 400 Bad Request.